src/vaadin-grid-scroller.js

```diff
@@ -0,0 +1,414 @@
+import { animationFrame } from '@polymer/polymer/lib/utils/async.js';
+import { flush } from '@polymer/polymer/lib/utils/flush.js';
+import { Debouncer } from '@polymer/polymer/lib/utils/debounce.js';
+import { PolymerIronList } from './iron-list.js';
+
+/**
+ * This Element is used internally by vaadin-grid.
+ *
+ * @implements PolymerIronList
+ * @extends HTMLElement
+ */
+class ScrollerElement extends PolymerIronList {
+  static get properties() {
+    return {
+      size: {
+        type: Number,
+        notify: true
+      },
+
+      /**
+       * @type {number}
+       * @protected
+       */
+      _vidxOffset: {
+        type: Number,
+        value: 0
+      }
+    };
+  }
+
+  static get observers() {
+    return ['_effectiveSizeChanged(_effectiveSize)'];
+  }
+
+  /** @protected */
+  connectedCallback() {
+    super.connectedCallback();
+    this._scrollHandler();
+  }
+
+  /**
+   * @param {!HTMLTableRowElement} item
+   * @param {number} index
+   * @protected
+   */
+  _updateScrollerItem() {}
+
+  /** @protected */
+  _afterScroll() {}
+
+  /** @protected */
+  _getRowTarget() {}
+
+  /** @protected */
+  _createScrollerRows() {}
+
+  /** @protected */
+  _canPopulate() {}
+
+  /**
+   * @param {number} index
+   * @protected
+   */
+  scrollToIndex(index) {
+    this._warnPrivateAPIAccess('scrollToIndex');
+
+    this._scrollingToIndex = true;
+    index = Math.min(Math.max(index, 0), this._effectiveSize - 1);
+    this.$.table.scrollTop = (index / this._effectiveSize) * (this.$.table.scrollHeight - this.$.table.offsetHeight);
+    this._scrollHandler();
+
+    if (this._accessIronListAPI(() => this._maxScrollTop) && this._virtualCount < this._effectiveSize) {
+      this._adjustVirtualIndexOffset(1000000);
+    }
+
+    this._accessIronListAPI(() => super.scrollToIndex(index - this._vidxOffset));
+    this._scrollHandler();
+
+    // Ensure scroll position
+    const row = Array.from(this.$.items.children).filter((child) => child.index === index)[0];
+    if (row) {
+      const headerOffset = row.getBoundingClientRect().top - this.$.header.getBoundingClientRect().bottom;
+      if (Math.abs(headerOffset) > 1) {
+        this.$.table.scrollTop += headerOffset;
+        this._scrollHandler();
+      }
+    }
+
+    this._scrollingToIndex = false;
+  }
+
+  /** @private */
+  _effectiveSizeChanged(size) {
+    let fvi; // first visible (adjusted) index
+    let fviOffset = 0;
+    this._iterateItems((pidx, vidx) => {
+      if (vidx === this._firstVisibleIndex) {
+        const row = this._physicalItems[pidx];
+        fvi = row.index;
+        fviOffset = row.getBoundingClientRect().top;
+      }
+    });
+
+    if (this.items && size < this.items.length) {
+      // Size was reduced, scroll to 0 first
+      this._scrollTop = 0;
+    }
+    if (!Array.isArray(this.items)) {
+      // Edge/IE seems to have the lowest maximum
+      const maxVirtualItems = this._edge ? 30000 : 100000;
```

Comment:
This limit no longer applies for Chromium Edge. 100000 can be used on all supported browsers.

Reply:
Done.